pkg/controller/servicebindingrequest/sbrcontroller.go

```diff
@@ -113,18 +113,32 @@ func updateFunc(logger *log.Log) func(updateEvent event.UpdateEvent) bool {
 		isConfigMap := isOfKind(e.ObjectNew, "ConfigMap")
 
 		if isSecret || isConfigMap {
-			dataFieldsAreEqual, err := compareObjectFields(e.ObjectNew, e.ObjectOld, "data")
+			dataFieldsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "data")
 			if err != nil {
 				logger.Error(err, "error comparing object fields: %s", err.Error())
 				return false
 			}
-
 			logger.Debug("Predicate evaluated", "dataFieldsAreEqual", dataFieldsAreEqual)
 			return !dataFieldsAreEqual
 		}
+		logger.Debug("updateFunc", "Predicate evaluated for Kind", e.ObjectOld.GetObjectKind)
```

Comment:
what the the added value of this message? Reading it in a log, I do not see any value. At this point, we still about to evaluate it.

Reply:
I can update the log to be something like `Predicate to be evaluated for Kind` . It will then provide info of the resource Kind for which the predicate is further evaluated.

Reply:
Referring now https://github.com/redhat-developer/service-binding-operator/pull/597/files#r482115743 Thanks for that suggestion

Reply:
Removed this